nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -559,4 +592,73 @@ public DocumentRef getCommentedDocumentRef(CoreSession session, DocumentModel co
 
         return commentedDocModel.getRef();
     }
+
+    /**
+     * Checks if a document has comments.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return {@code true} if comments were found
+     * @since 11.1
+     */
+    protected boolean hasComments(CoreSession session, DocumentModel document) {
+        String query = String.format( //
+                QUERY_GET_COMMENTS_UUID_BY_COMMENT_ANCESTOR, document.getId());
+        return !session.queryProjection(query, 1, 0).isEmpty();
+    }
+
+    /**
+     * Checks if a document has comments from a particular user.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @param user the name of the user who's comments are being counted
+     * @return {@code true} if comments by user were found
+     * @since 11.1
+     */
+    protected boolean hasComments(CoreSession session, DocumentModel document, String user) {
+        String query = String.format( //
+                QUERY_GET_COMMENTS_UUID_BY_COMMENT_ANCESTOR_AND_AUTHOR, document.getId(), user);
+        return !session.queryProjection(query, 1, 0).isEmpty();
+    }
+    protected void handleNotificationAutoSubscriptions(CoreSession session, DocumentModel commentDocModel,
```

Comment:
add a new line, between the two methods

Reply:
Could you rework the method signature  to this?
```java
void handleNotificationAutoSubscriptions(CoreSession session, DocumentModel topLevelDocModel, DocumentModel commentDocModel)
```
In order to follow same order than existing one and have a better naming for top level.